lib/vtls/openssl.c

```diff
@@ -2888,8 +2888,7 @@ static CURLcode ossl_connect_step1(struct Curl_easy *data,
   if((SSL_CONN_CONFIG(verifypeer) || SSL_CONN_CONFIG(verifyhost)) &&
      (SSL_SET_OPTION(native_ca_store))) {
     X509_STORE *store = SSL_CTX_get_cert_store(backend->ctx);
-    HCERTSTORE hStore = CertOpenSystemStore((HCRYPTPROV_LEGACY)NULL,
-                                            TEXT("ROOT"));
+    HCERTSTORE hStore = CertOpenSystemStore(0, TEXT("ROOT"));
```

Comment:
This is curious I think as the [docs](https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certopensystemstorea) says:

> This parameter is not used and should be set to **NULL**

So, the removed typecast is one thing but is this a pointer or not a pointer?

Reply:
`HCRYPTPROV_LEGACY` is a typedef for `ULONG_PTR`, which is an integer type that's the size of a pointer, so it seems to make more sense to use 0 instead.

Reply:
Wait, `ULONG_PTR` is not a pointer? I just have to double-check this!